From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martin Wilck <mwilck@suse.com>
Date: Fri, 28 Oct 2022 15:29:51 +0200
Subject: [PATCH] GitHub workflows: use make -j8 -Orecurse

Without -Orecurse, quiet make output is confusing in parallel mode
because make prints "Entering directory" and "Leaving directory"
messages before and after every target.

Use parallel compilation also in native.yaml and foreign.yaml.

Signed-off-by: Martin Wilck <mwilck@suse.com>
Signed-off-by: Benjamin Marzinski <bmarzins@redhat.com>
---
 .github/workflows/abi.yaml                |  2 +-
 .github/workflows/build-and-unittest.yaml | 18 +++++++++---------
 .github/workflows/coverity.yaml           |  2 +-
 .github/workflows/foreign.yaml            |  4 ++--
 .github/workflows/native.yaml             |  6 +++---
 5 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/.github/workflows/abi.yaml b/.github/workflows/abi.yaml
index 89b971cd..644fcb62 100644
--- a/.github/workflows/abi.yaml
+++ b/.github/workflows/abi.yaml
@@ -32,7 +32,7 @@ jobs:
           libdevmapper-dev libreadline-dev libaio-dev libsystemd-dev
           libudev-dev libjson-c-dev liburcu-dev libcmocka-dev libedit-dev
       - name: create ABI
-        run: make -O -j$(grep -c ^processor /proc/cpuinfo) abi.tar.gz
+        run: make -Orecurse -j$(grep -c ^processor /proc/cpuinfo) abi.tar.gz
       - name: save ABI
         uses: actions/upload-artifact@v1
         with:
diff --git a/.github/workflows/build-and-unittest.yaml b/.github/workflows/build-and-unittest.yaml
index a5a0717d..8b6bb776 100644
--- a/.github/workflows/build-and-unittest.yaml
+++ b/.github/workflows/build-and-unittest.yaml
@@ -32,11 +32,11 @@ jobs:
           libdevmapper-dev libreadline-dev libaio-dev libsystemd-dev
           libudev-dev libjson-c-dev liburcu-dev libcmocka-dev libedit-dev
       - name: build
-        run: make -O -j$(grep -c ^processor /proc/cpuinfo) READLINE=${{ matrix.rl }}
+        run: make -Orecurse -j$(grep -c ^processor /proc/cpuinfo) READLINE=${{ matrix.rl }}
       - name: test
-        run: make -O -j$(grep -c ^processor /proc/cpuinfo) test
+        run: make -Orecurse -j$(grep -c ^processor /proc/cpuinfo) test
       - name: valgrind-test
-        run: make -O -j$(grep -c ^processor /proc/cpuinfo) valgrind-test
+        run: make -Orecurse -j$(grep -c ^processor /proc/cpuinfo) valgrind-test
       - name: valgrind-results
         run: cat tests/*.vgr
       - name: clean-nonroot-artifacts
@@ -65,11 +65,11 @@ jobs:
       - name: set CC
         run: echo CC=gcc-10 >> $GITHUB_ENV
       - name: build
-        run: make -O -j$(grep -c ^processor /proc/cpuinfo) READLINE=${{ matrix.rl }}
+        run: make -Orecurse -j$(grep -c ^processor /proc/cpuinfo) READLINE=${{ matrix.rl }}
       - name: test
-        run: make -O -j$(grep -c ^processor /proc/cpuinfo) test
+        run: make -Orecurse -j$(grep -c ^processor /proc/cpuinfo) test
       - name: valgrind-test
-        run: make -O -j$(grep -c ^processor /proc/cpuinfo) valgrind-test
+        run: make -Orecurse -j$(grep -c ^processor /proc/cpuinfo) valgrind-test
       - name: valgrind-results
         run: cat tests/*.vgr
       - name: clean-nonroot-artifacts
@@ -98,11 +98,11 @@ jobs:
       - name: set CC
         run: echo CC=clang >> $GITHUB_ENV
       - name: build
-        run: make -O -j$(grep -c ^processor /proc/cpuinfo) READLINE=${{ matrix.rl }}
+        run: make -Orecurse -j$(grep -c ^processor /proc/cpuinfo) READLINE=${{ matrix.rl }}
       - name: test
-        run: make -O -j$(grep -c ^processor /proc/cpuinfo) test
+        run: make -Orecurse -j$(grep -c ^processor /proc/cpuinfo) test
       - name: valgrind-test
-        run: make -O -j$(grep -c ^processor /proc/cpuinfo) valgrind-test
+        run: make -Orecurse -j$(grep -c ^processor /proc/cpuinfo) valgrind-test
       - name: valgrind-results
         run: cat tests/*.vgr
       - name: clean-nonroot-artifacts
diff --git a/.github/workflows/coverity.yaml b/.github/workflows/coverity.yaml
index 3c6b3824..321b94e0 100644
--- a/.github/workflows/coverity.yaml
+++ b/.github/workflows/coverity.yaml
@@ -32,7 +32,7 @@ jobs:
       - name: build with cov-build
         run: >
           PATH="$PWD/coverity/bin:$PATH"
-          cov-build --dir cov-int make -O -j"$(grep -c ^processor /proc/cpuinfo)"
+          cov-build --dir cov-int make -Orecurse -j"$(grep -c ^processor /proc/cpuinfo)"
       - name: pack results
         run: tar cfz multipath-tools.tgz cov-int
       - name: submit results
diff --git a/.github/workflows/foreign.yaml b/.github/workflows/foreign.yaml
index 5a19913a..72ac24fb 100644
--- a/.github/workflows/foreign.yaml
+++ b/.github/workflows/foreign.yaml
@@ -21,11 +21,11 @@ jobs:
         uses: actions/checkout@v1
       - name: build and test
         if: ${{ matrix.arch == '' || matrix.arch == '-i386' }}
-        run: make test
+        run: make -j8 -Orecurse test
       - name: build
         if: ${{ matrix.arch != '' && matrix.arch != '-i386' }}
         # The build path is different between builder and runner
-        run: make TESTDIR=${{ github.workspace }}/tests test-progs
+        run: make -j8 -Orecurse TESTDIR=${{ github.workspace }}/tests test-progs
       - name: archive
         if: ${{ matrix.arch != '' && matrix.arch != '-i386' }}
         run: >
diff --git a/.github/workflows/native.yaml b/.github/workflows/native.yaml
index 8b599209..68bf0984 100644
--- a/.github/workflows/native.yaml
+++ b/.github/workflows/native.yaml
@@ -92,10 +92,10 @@ jobs:
       - name: checkout
         uses: actions/checkout@v1
       - name: build and test
-        run: make test
+        run: make -j8 -Orecurse test
       - name: clean
-        run: make clean
+        run: make -j8 -Orecurse clean
       - name: clang
         env:
           CC: clang
-        run: make test
+        run: make -j8 -Orecurse test
